### ISSUE TITLE
john.conf ShowUIDinCracks = Bool

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#517]
Forgotten item from http://www.openwall.com/lists/john-dev/2013/01/30/16 (see following posts too).

We could add a john.conf option ShowUIDinCracks = Bool, that when set will add the uid to the crack output. So instead of the normal real-time crack output:
```
password123      (Administrator)
sesame           (root)
Induction        (netgear)
```
We'll get this:
```
password123      (Administrator:500)
sesame           (root:0)
Induction        (netgear:31-33-7b-ab-e5-00)
```
...or something like that. Something similar could be done to --show using the same config option.

**Comment by jfoug** [comment]
since the login for split hashes is already user:1 user:2, the final output will end up being the former format:

d123             (Administrator:2:500)
passwor          (Administrator:1:500)

I have a couple things to fix up (such as getting -show to work, and making sure that -show=left  works, I think it should).   This was not overly hard to do, but it did get a little intrusive, and required a couple structure changes. 

Right now, this method is honored on screen output, in the .log file, and I think in -show=left.  I have to find where to put it in -show, and once there I should be done.

**Comment by jfoug** [comment]
-show=left not quite right (wrong order).

**Comment by jfoug** [comment]
-show already has FULL line data (uid, gid, and I think everything off the line). It does not 'honor' the ShowUIDinCracks = Bool  but I really do not care.

So now, if ShowUIDinCracks = Bool is set, you see the uid on crack, -show=left, in the .log file, and on -show.   Can you think of anywhere this has been missed?  I will get code checked in shortly.  I believe it is solid.

NOTE, did we ever get *cap2john to output the MAC address for the uid ?

**Comment by jfoug** [comment]
I think this is now completed.  But I will wait to hear from others.

**Comment by magnumripper** [comment]
Looks good to me, I will test a little.

**Comment by magnumripper** [comment]
Pwdump format bug fixed in 44500f1, field separator bugs fixed in 151bcd2. The order of --show=left looks correct to me (although the order of pwdump files are always a bit weird). Closing.
